src/Compilers/CSharp/Portable/Declarations/DeclarationTreeBuilder.cs

```diff
@@ -74,6 +75,11 @@ private DeclarationTreeBuilder(SyntaxTree syntaxTree, string scriptClassName, bo
                     firstGlobalStatement ??= global;
                     var topLevelStatement = global.Statement;
 
+                    if (!topLevelStatement.IsKind(SyntaxKind.EmptyStatement))
+                    {
+                        hasNonEmptyGlobalSatement = true;
+                    }
```

Comment:
Should the error be reported if the top-level program contains only local functions?

Reply:
> Should the error be reported if the top-level program contains only local functions?

[https://github.com/dotnet/roslyn/issues/53472#issuecomment-866502606](https://github.com/dotnet/roslyn/issues/53472#issuecomment-866502606) - "LDM decided to make it an error if all top-level statements are empty." I do not see LDM deciding anyjthing about local function statements.

Reply:
@AlekseyTs Yeah I see the PR matches LDM decision, but I mean, was that considered?

The goal of the error is to tell the user "your program will execute nothing and just return 0 immediately", which is the case when the top-level statements are local functions.

Reply:
> The goal of the error is to tell the user "your program will execute nothing and just return 0 immediately", 

I do not think that is the goal, an entry point function is always executed. There is plenty of ways to write code that does nothing even when it has non-empty statements. We were addressing specific customers scenario and I don't think there was a goal to generalize the issue. If you feel that the issue should be generailized, open an issue and provide your reasons. I beleive a local function that is never called produces a warning, in my opinion that is goog enough to give a hint that the code probably doesn't do what was intended.